tests/ut/test_core_setup.py

```diff
@@ -0,0 +1,268 @@
+from unittest import mock
+
+from cterasdk import toxmlstr
+from cterasdk import exception
+from cterasdk.core import setup
+from cterasdk.core.enum import ServerMode, SlaveAuthenticaionMethod, SetupWizardStage, SetupWizardStatus
+from cterasdk.common import Object
+from tests.ut import base_core
+
+
+class TestCoreSetup(base_core.BaseCoreTest):  # pylint: disable=too-many-instance-attributes
+
+    def setUp(self):
+        super().setUp()
+        self._master_ipaddr = '8.8.8.8'
+        self._master_secret = 'password'
+        self._admin_username = 'admin'
+        self._admin_email = 'admin@ctera.com'
+        self._admin_first_name = 'Service'
+        self._admin_last_name = 'Account'
+        self._admin_password = 'password'
+        self._domain = 'ctera.me'
+        self._replication_candidates = ['objs/6//Server/server', 'objs/7//Server/server1', 'objs/8//Server/server3']
+        self._replicate_from = 'server1'
+        self._array_of_statuses = []
```

Comment:
Actually you don't need this array or the method that uses it. You can set the mock's side_effect to be the array and it will do exactly what you've implemented (See https://stackoverflow.com/questions/24897145/python-mock-multiple-return-values)